metagraph/src/graph/annotated_dbg.cpp

```diff
@@ -348,6 +350,11 @@ ::call_annotated_nodes(const std::string &label,
 }
 
 bool AnnotatedSequenceGraph::check_compatibility() const {
+    if (dynamic_cast<const CanonicalDBG*>(graph_.get())) {
+        return dynamic_cast<const CanonicalDBG&>(*graph_).get_graph().max_index()
+            == annotator_->num_objects();
+    }
```

Comment:
maybe we can redefine max index: let it return maximum node index in map_to_nodes, that is, only for annotated nodes.
Then we wouldn't need to do this hack

Reply:
Maybe a cleaner approach would be to add a method `max_annotated_index` to `AnnotatedDBG`

Reply:
That'll cause confusion, though, since the max index in a wrapped graph will no longer be correct.

Maybe a less confusion approach (which would in effect be moving this hack somewhere else) would be to add a method `max_annotated_index` to `AnnotatedDBG`

Reply:
It should definitely be in the same class as `map_to_nodes`. It's better if they both move to AnnotatedDBG, but this is not a completely trivial refactoring. So I think for now it's better to keep them both in graphs.

Reply:
Turns out, `max_index()` is already used in many places as a bound on indexes in `map_to_nodes_sequentially`, so we can't easily redefine it.

In the future, we need to add another method that returns max index called in `map_to_nodes`.